app/src/main/java/com/github/geohunt/app/ui/components/challenge/MainUserView.kt

```diff
@@ -122,3 +127,32 @@ internal fun MainUserView(challenge: Challenge) {
     }
 }
 
+@Composable
+internal fun LoggedUserContext.FollowButton(author: User, modifier: Modifier = Modifier) {
+    val doesFollow = rememberLazyRef {
+        author.doesFollow
+    }
+
+    Button(
+        modifier = modifier,
+        contentPadding = PaddingValues(2.dp, 2.dp),
+        shape = RoundedCornerShape(12.dp),
+        enabled = doesFollow.value != null,
+        onClick = {
+            if (author.doesFollow.value != null) {
+                if (author.doesFollow.value!!) {
+                    author.unfollow()
+                } else {
+                    author.follow()
+                }
+            }

```

Comment:
In the onClick lambda, it is assumed that doesFollow.value is not null. This is not always the case, since the LazyRef object might not have been initialized yet. It would be better to first check if doesFollow.value is not null before accessing it.